engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/ExecutionEntity.java

```diff
@@ -774,9 +783,13 @@ public void setProcessDefinition(ProcessDefinitionImpl processDefinition) {
     this.processDefinition = processDefinition;
     if (processDefinition != null) {
       this.processDefinitionId = processDefinition.getId();
+      if (processDefinition instanceof ProcessDefinitionEntity) {
```

Comment:
📓 Process definition can be created without a key. Therefore, setting the key only if we have have it.
Found out about the use case via failing `org.camunda.bpm.engine.test.standalone.entity.EntitySerializationTest.testExecutionEntitySerialization()`.